### PR TITLE
Support pulls/{number}/commits

### PR DESCRIPTION
**Pull request by SamWilsn**

### CODE REVIEW

**Comment by XAMPPRocky** [comment]
Thank you for your PR! Can you rebase it on top on main?

**Comment by SamWilsn** [comment]
Done!

**Comment by SamWilsn** [comment]
@XAMPPRocky So it seems like GitHub's schema isn't entirely correct for the `Author` type, or maybe I'm doing something wrong. For at least one pull request, I'm getting back JSON that's deserializing to `Author` that doesn't have `login` or `id`. Have you encountered this before?

**Comment by XAMPPRocky** [comment]
@SamWilsn Yes, GitHub's schema is descriptive at best. It doesn't match the reality of the API and what the API will actually return.

**Comment by SamWilsn** [comment]
Should I change the `Author` type to have optional fields, or make a new type specific to this endpoint?

**Comment by XAMPPRocky** [comment]
Make it have optional fields, there might other endpoints where this also happens.

**Comment by SamWilsn** [comment]
It's, uh... mostly `Option` now :sweat_smile:

File: src/models.rs [hunk 1]
```diff
@@ -398,24 +398,42 @@ pub struct IssuePullRequest {
...
-    pub starred_url: Url,
-    pub subscriptions_url: Url,
-    pub organizations_url: Url,
-    pub repos_url: Url,
-    pub events_url: Url,
-    pub received_events_url: Url,
-    pub r#type: String,
-    pub site_admin: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub login: Option<String>,
```
[the remark this answers is not in the document]
Comment (SamWilsn) [reply src/models.rs:402]: I can't provide a schema, but here's an API response:

https://api.github.com/repos/ethereum/ERCs/commits/fb8277911c110491bf3610b5e0298c4f4cb70a3f

[the remark this answers is not in the document]
Comment (SamWilsn) [reply src/models.rs:402]: I don't think so. On this commit we have:

```js
{
  // ...
  "comments_url": "https://api.github.com/repos/ethereum/ERCs/commits/fb8277911c110491bf3610b5e0298c4f4cb70a3f/comments",
  "author": {

  },
  "committer": {

  },
  "parents": [ /* ... */ ],
  // ...
}
```

Note the empty (but non-null) `committer` and `author` items.

---

On [another commit](https://api.github.com/repos/ethereum/ERCs/commits/ce626a0f5d26545ececf4ca6ea4bf64c16b6b233) we have:

```js
{
  // ...
  "comments_url": "https://api.github.com/repos/ethereum/ERCs/commits/ce626a0f5d26545ececf4ca6ea4bf64c16b6b233/comments",
  "author": {
    "login": "tokernomics",
    "id": 88177870,
    "node_id": "MDQ6VXNlcjg4MTc3ODcw",
    "avatar_url": "https://avatars.githubusercontent.com/u/88177870?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/tokernomics",
    "html_url": "https://github.com/tokernomics",
    "followers_url": "https://api.github.com/users/tokernomics/followers",
    "following_url": "https://api.github.com/users/tokernomics/following{/other_user}",
    "gists_url": "https://api.github.com/users/tokernomics/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/tokernomics/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/tokernomics/subscriptions",
    "organizations_url": "https://api.github.com/users/tokernomics/orgs",
    "repos_url": "https://api.github.com/users/tokernomics/repos",
    "events_url": "https://api.github.com/users/tokernomics/events{/privacy}",
    "received_events_url": "https://api.github.com/users/tokernomics/received_events",
    "type": "User",
    "site_admin": false
  },
  "committer": {
    "login": "web-flow",
    "id": 19864447,
    "node_id": "MDQ6VXNlcjE5ODY0NDQ3",
    "avatar_url": "https://avatars.githubusercontent.com/u/19864447?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/web-flow",
    "html_url": "https://github.com/web-flow",
    "followers_url": "https://api.github.com/users/web-flow/followers",
    "following_url": "https://api.github.com/users/web-flow/following{/other_user}",
    "gists_url": "https://api.github.com/users/web-flow/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/web-flow/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/web-flow/subscriptions",
    "organizations_url": "https://api.github.com/users/web-flow/orgs",
    "repos_url": "https://api.github.com/users/web-flow/repos",
    "events_url": "https://api.github.com/users/web-flow/events{/privacy}",
    "received_events_url": "https://api.github.com/users/web-flow/received_events",
    "type": "User",
    "site_admin": false
  },
  "parents": [ /* ... */ ],
  // ...
}
```

The data here more closely matches the `Author` struct, and not the `GitUser` struct.

[the remark this answers is not in the document]
Comment (SamWilsn) [reply src/models.rs:402]: I am not familiar enough with the GitHub API to judge. I'm just going off [this comment](https://github.com/XAMPPRocky/octocrab/pull/656#issuecomment-2179495933).

[the remark this answers is not in the document]
Comment (SamWilsn) [reply src/models.rs:402]: Seems like it's been broken for a while: https://github.com/orgs/community/discussions/57497

**Comment by SamWilsn** [comment]
Seems like GitHub hasn't gotten around to fixing either the documentation or the API endpoint. How would you like to proceed?

**Comment by SamWilsn** [comment]
> I think they did fix the docs.

I don't know how I missed that! I'm sorry.



> Would it be possible to write code to conditionally deserialize into either `Author` (with fields), or into an empty dictionary (without any fields)?

I could do something like:

```rust
#[derive(Deserialize)]
#[serde(deny_unknown_fields)]
struct Empty {
}

#[derive(Deserialize)]
#[serde(untagged)]
enum MaybeUser {
    Empty(Empty),
    Author(Author),
}
```

**Comment by XAMPPRocky** [comment]
I think it would be better to have it be an option, if it's empty it should be considered equivalent to null.

**Comment by SamWilsn** [comment]
@XAMPPRocky @maflcko Sorry for the super long delay on this one. Noticed you've added support for pr commits! I've co-opted this pull request into adding support for deserializing `{}`.

Hopefully I've followed the specific pr style well enough!

**Comment by XAMPPRocky** [comment]
Thank you for your PR, and congrats on your first contribution! 🎉

**Pull request merged**